org/mozilla/jss/netscape/security/x509/AlgorithmId.java

```diff
@@ -177,12 +189,88 @@ public static AlgorithmId parse(byte[] val)
     }
 
     /**
-     * Constructs a parameterless algorithm ID.
+     * Constructs an algorithm ID with a fully encoded params object
+     *
+     * @ param oid the identifier for the algorithm
+     * @ param params the fully encoded AlgorithmIdentifier Object
+     */
+
+    public AlgorithmId(ObjectIdentifier oid, AlgorithmParameters params)
+        throws IOException {
+       algid = oid;
+       algParams = params;
+
+       if(algParams != null ) {
+           this.params = new DerValue(algParams.getEncoded());
+       }
+    }
+
+    /**
+     * Constructor that takes the oid and name, so the name can be cachedf or laster use.
+     * @throws NoSuchAlgorithmException
+     * @throws IOException
+     *
+     */
+
+    public AlgorithmId(ObjectIdentifier oid, String algName) throws  IOException {
```

Comment:
Remove space above, and should this also throw NSAE?

Reply:
Done.